python/ray/_raylet.pyx

```diff
@@ -1624,6 +1622,11 @@ cdef class CoreWorker:
                             return_ids_vector, return_ids[i]))
                     return_ids_vector.clear()
 
+            with nogil:
```

Comment:
Isn't this breaking Java implementation?

Reply:
Maybe we should change the Java code path as well?

Reply:
Oh yea, this would break the C++ and Java runtime. I should fix those too.

Reply:
Ah I think you're right. I thought we could get away with doing this incrementally, but I guess we need to modify the other language bindings too. @franklsf95 can you take care of this?